src/perform_step/adams_nordsieck_perform_step.jl

```diff
@@ -120,24 +134,22 @@ end
 
   ################################### Error estimation
 
-  #if integrator.opts.adaptive
-  #  utilde = cache.Δ*cache.tq
-  #  atmp = calculate_residuals(utilde, uprev, u, integrator.opts.abstol, integrator.opts.reltol,integrator.opts.internalnorm)
-  #  integrator.EEst = integrator.opts.internalnorm(atmp)
-  #end
+  if integrator.opts.adaptive
+    @. tmp = const_cache.Δ*const_cache.tq
+    calculate_residuals!(atmp, tmp, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm)
+    integrator.EEst = integrator.opts.internalnorm(atmp)
+    if integrator.EEst >= one(integrator.EEst)
+      perform_predict!(cache, true)
```

Comment:
what's this?

Reply:
This is to rewind the Nordsieck vector back if the error test failed.

Reply:
We definitely need more checkings though. I will do it in the next PR.